### DISCUSSION TITLE
ERROR:tweepy.streaming:Stream encountered HTTP error: 403"  please help

### DISCUSSION THREAD

**Discussion by yunsanch** [tweepy/tweepy#1947] [Questions] [answered]
hello I am having problems getting this error HTTP 403 after running this code below with the class FavRetweetListener, can someone please guide me to the best possible way to fix that error? I have a Twitter developer elevated access account, please anything that can guide me to the right place to get it fixed or any suggested solutions will be greatly appreciated it.

INFO:root: API created
ERROR:tweepy.streaming:Stream encountered HTTP error: 403"
#this is the config file i have

#tweepy-bots/bots/config.py

import tweepy

import logging

import os

logger = logging.getLogger()

def create_api():

    consumer_key = "x"
    consumer_secret = "x"
    access_token = "x"
    access_token_secret = "x"
    bearer_token = "x"

    auth = tweepy.OAuth1UserHandler(consumer_key, 
consumer_secret,access_token,access_token_secret)
# auth.set_access_token(access_token,access_token_secret)
    auth.secure = True
    api = tweepy.API(auth,wait_on_rate_limit=True)

    try:
        api.verify_credentials()
    
    except Exception as e:
        print("there is a problem")
        logger.error("Error creating API",exc_info=True)
        raise e
    logger.info("API created")
    return api
""" and here below is the code I am running that gives me the error above """

from multiprocessing connection import Listener

from config import create_api

import tweepy

import logging

from textblob import TextBlob

import json

consumer_key = "x"
consumer_secret = "x"
access_token = "x"
access_token_secret = "x"
bearer_token = "x"

logging.basicConfig(level=logging.INFO)

logger = logging.getLogger()

Class FavRetweetListeener(tweepy.StreamingClient):

    def __init__(self,api):
        self.api = api

    # me =self.api.me()
        self.user_id = api.verify_credentials()
    # self.user = api.verify_credentials()
    
    def on_status(self, tweet):
    # bot_id= int(api.me().id_str())
        tweet_id = tweet.id
        logger.info(f"Processing tweet id {tweet_id}")
    # tweet_analysis = TextBlob(tweet.text)
    # tweet_analysis_score = tweet_analysis.sentiment.polarity

    # if tweet.in_reply_to_status_id is not None or tweet.author.id == bot_id and tweet_analysis_score >= 0.2 and not tweet.retweeted:

        if tweet.in_reply_to_status_id is not None or tweet.user_id == self.user_id:
        #this tweet is a reply or i am its author so, ignore it
            return
        if not tweet.favorited:
        #mark it as liked, since we have not done it yet
           try:
                tweet.get_favorites()
           except Exception as e:
                logger.error("Error on fav",exc_info=True)
        if not tweet.retweeted:
            try:
                tweet.get_retweets()
            except Exception as e:
                logger.error("Error on fav and retweet",exc_info=True)

    def on_error(self, status):
        logger.error(status)

def main(keywords):

    api = create_api()

    tweets_listener = FavRetweetListeener(api)

    stream = tweepy.Stream(consumer_key,consumer_secret,access_token,access_token_secret)

    stream = tweepy.Stream(auth=api.auth,Listener= tweets_listener)

    stream.filter(track=keywords, languages=["en"])  

if __name__ == "__main__":

    main(["MLB","Python"])

**Comment by caretak3r** [comment]
Please post the output from the execution. 401 indicates that your credentials are incorrect. Can you confirm this?

**Comment by yunsanch** [comment]
![Screenshot 2022-07-08 191951](https://user-images.githubusercontent.com/60577975/178081374-c9e24fd3-eb89-4d86-8229-7b7a1db35bba.png)

**Comment by yunsanch** [comment]
my credential should be correct since I use the same credential in the code below and I have elevated access 
    #!/usr/bin/env python
    #tweepy-bots/bots/followfollowers.py

    import tweepy
    import logging
    from config import create_api
    import time

    logging.basicConfig(level=logging.INFO)
    logger =logging.getLogger()

    def follow_followers(api):
        logger.info("Retrieving and following followers")
        for follower in tweepy.Cursor(api.get_followers).items(): #had to changed get followers since followers was not longer useif 
    not follower.following:
                logger.info(f"Following {follower.name}")
                follower.follow()

    def main():
        api =create_api()
        while True:
            follow_followers(api)
            logger.info("Waiting..")
            time.sleep(60)

    if __name__ == "__main__":
        main()

**Comment by yunsanch** [comment]
the above code works file

**Comment by Harmon758** [comment]
What version of Tweepy are you using?
Are you using a Twitter Developer App created on or after 2022-04-29?

For code block usage, see https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.

**Comment by yunsanch** [comment]
hello @Harmon758 i am using 4.10.0, i am uing a twiter developer app created on July, i also have both regular and elevated access anable, see picture below
![image](https://user-images.githubusercontent.com/60577975/185745695-8614964d-c281-4700-a3f8-c45b48c8d9f6.png)

**Comment by Harmon758** [comment, marked as answer]
Then this is a duplicate of #1877 / #1878.

> New Apps:
> > will not be able to gain access to v1.1 statuses/sample and v1.1 statuses/filter
> 
> https://twittercommunity.com/t/deprecation-announcement-removing-compliance-messages-from-statuses-filter-and-retiring-statuses-sample-from-the-twitter-api-v1-1/170500
> 
> You'll have to use Twitter API v2.

**Comment by yunsanch** [comment]
i still continue to have an error dispise haveing my code set from Stream to StreamingClient i still get the "Stream encounter HTTP error: 403"

**Reply by Harmon758** [reply]
What's the relevant code?

**Reply by yunsanch** [reply]
`#!/usr/bin/env python
# tweepy-bots/bots/favretweet.py

from multiprocessing.connection import Listener
from config import create_api
import tweepy
import logging
from textblob import TextBlob
import json

consumer_key = "x"
consumer_secret = "x"
access_token = "x"
access_token_secret = "x"
bearer_token = "x"

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger()

class FavRetweetListeener(tweepy.StreamingClient):
    def __init__(self,api):
        self.api = api
        # me =self.api.me()
        self.verify_credentials = api.verify_credentials()
        # self.user_id = api.verify_credentials()
        # self.user = api.verify_credentials()
        
    def on_status(self, tweet):
        # bot_id= int(api.me().id_str())
        tweet_id = tweet.id
        logger.info(f"Processing tweet id {tweet_id}")
        # tweet_analysis = TextBlob(tweet.text)
        # tweet_analysis_score = tweet_analysis.sentiment.polarity

        # if tweet.in_reply_to_status_id is not None or tweet.author.id == bot_id and tweet_analysis_score >= 0.2 and not tweet.retweeted:

        if tweet.in_reply_to_status_id is not None or tweet.user_id == self.verify_credentials.user_id:
            #this tweet is a reply or i am its author so, ignore it
            return
        if not tweet.favorited:
            #mark it as liked, since we have not done it yet
            try:
                tweet.get_favorites()
            except Exception as e:
                logger.error("Error on fav",exc_info=True)
        if not tweet.retweeted:
            try:
                tweet.get_retweets()
            except Exception as e:
                logger.error("Error on fav and retweet",exc_info=True)

    def on_error(self, status):
        logger.error(status)


def main(keywords):
    api = create_api()
    tweets_listener = FavRetweetListeener(api)
    stream = tweepy.StreamingClient(bearer_token,)
    # stream = tweepy.StreamingClient(consumer_key,consumer_secret,access_token,access_token_secret)
    # stream = tweepy.Stream(auth=api.auth,Listener= tweets_listener)
    stream.filter(track=keywords, languages=["en"])
if __name__ == "__main__":
    main(["MLB","Python"])`

**Reply by yunsanch** [reply]
![image](https://user-images.githubusercontent.com/60577975/185749749-c0fe8355-a6b8-47b5-b3a8-c8d60d74276f.png)
getting different error now

**Reply by Harmon758** [reply]
Check that your credentials are correct and that your clock is synced.
https://developer.twitter.com/en/docs/twitter-api/v1/tweets/filter-realtime/guides/connecting

Again, for code block usage, see https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.
